datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -1311,44 +1197,25 @@ mod tests {
         assert_eq!(col("c1").get_type(&schema).unwrap(), DataType::Utf8);
 
         assert_eq!(
-            do_simplify(col("c1").not_eq(lit(true))),
-            col("c1").not_eq(lit(true)),
-        );
-
-        assert_eq!(
```

Comment:
this was removed for the same reason as the tests above - they don't make sense to compare a string column to a boolean constant (it will get a run time error)